src/main/java/it/gov/pagopa/pu/bff/connector/debt_position/client/ReceiptClient.java

```diff
@@ -0,0 +1,61 @@
+package it.gov.pagopa.pu.bff.connector.debt_position.client;
+
+import it.gov.pagopa.pu.bff.connector.debt_position.config.DebtPositionApisHolder;
+import it.gov.pagopa.pu.bff.util.PageUtils;
+import it.gov.pagopa.pu.debtpositions.dto.generated.PagedModelReceiptView;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.time.OffsetDateTime;
+
+@Service
+@Slf4j
+public class ReceiptClient {
+
+  private final DebtPositionApisHolder debtPositionApisHolder;
+
+  public ReceiptClient(DebtPositionApisHolder debtPositionApisHolder) {
+    this.debtPositionApisHolder = debtPositionApisHolder;
+  }
+
+  @SuppressWarnings("squid:S107")
```

Comment:
can you introduce a DTO to keep together all filters? this way you could eliminate this suppressWarning (suppressWarning is generally forbidden! only a few useCase, queries, could allow it)

Reply:
The DTO should be introduced only on internal usage, not for the query parameter! actually you are forcing to pass a json through query param